service/core/action/post/publish.go

```diff
@@ -89,17 +72,19 @@ func publish(w http.ResponseWriter, r *http.Request) {
 		PostID: uint(id),
 	}).Count(&totAuthors)
 
+	tx := config.DB.Begin()
+
 	if totAuthors == 0 {
-		loggerx.Error(errors.New("cannot publish post without authors"))
-		errorx.Render(w, errorx.Parser(errorx.CannotSaveChanges()))
-		return
+		author := model.PostAuthor{
+			AuthorID: uint(uID),
+			PostID:   uint(id),
+		}
+		tx.Model(&model.PostAuthor{}).Create(&author)
 	}
 
-	tx := config.DB.Begin()
-
 	err = tx.Model(&result.Post).Set("gorm:association_autoupdate", false).Updates(model.Post{
 		Status:        "published",
-		PublishedDate: publish.PublishedDate,
+		PublishedDate: time.Now(),
 	}).Preload("Medium").Preload("Format").Preload("Tags").Preload("Categories").First(&result.Post).Error
 
```

Comment:
This query might return null instead of empty array incase of no post tags or categories. Once check it & make changes accordingly.

Reply:
done changes